### PR TITLE
schutzbot: Replace EL8 nightly pipeline with conditional steps

### PR DESCRIPTION
**Pull request by atodorov**
- looks like having 2 pipelines defined in the same Jenkinsfile
  is not working as we expect
- remove the existing EL8 nightly pipeline definition
- use detect_build_cause() funtion and set a global env.BUILD_CAUSE
  variable for use in conditionals
- add a cron job trigger - this will work together with the
  GitHub pull request trigger
- use conditional blocks for all steps we want to be executed
  outside of cron jobs
- only EL8.3 jobs will be executed unconditionally, both in cron
  and for PRs. The preparation stage for cron jobs makes sure to
  use the same name for osbuild-mock.repo so that the jobs can
  unstash it later!

### CODE REVIEW

**Comment by atodorov** [comment]
@major FYI this surprisingly works even for pull requests:
https://osbuildci.cloud.paas.psi.redhat.com/blue/organizations/jenkins/osbuild%2Fosbuild-composer/detail/PR-976/2/pipeline/ is an execution triggered after I refreshed the initial PR, some stages timed out for some reason.

Then 6 hours later we've got 
https://osbuildci.cloud.paas.psi.redhat.com/blue/organizations/jenkins/osbuild%2Fosbuild-composer/detail/PR-976/3/pipeline/
which says "Started by timer" and failed b/c I had forgotten a label section in the new stage.

**Comment by atodorov** [comment]
The most recent cron job from last night:
https://osbuildci.cloud.paas.psi.redhat.com/blue/organizations/jenkins/osbuild%2Fosbuild-composer/detail/PR-976/8/

Failure should be resolved by the latest commit.

File: schutzbot/prepare-rhel-nightly.sh [hunk 1]
```diff
@@ -52,3 +52,59 @@ tee rhel-8.json << EOF
...
+
+# Public URL for the S3 bucket with our artifacts.
+MOCK_REPO_BASE_URL="http://osbuild-composer-repos.s3-website.us-east-2.amazonaws.com"
+
+# Full URL to the RPM repository after they are uploaded.
+REPO_URL=${MOCK_REPO_BASE_URL}/${REPO_DIR_LATEST}
+
+# Ammend repository file.
+greenprint "📜 Ammend dnf repository file"
+tee -a osbuild-mock.repo << EOF
```
Comment (ondrejbudai) [inline schutzbot/prepare-rhel-nightly.sh:102]:
```suggestion
tee osbuild-mock.repo << EOF
```

Append is imho not needed here.

File: schutzbot/prepare-rhel-nightly.sh [hunk 2]
```diff
@@ -52,3 +52,59 @@ tee rhel-8.json << EOF
...
+greenprint "☁ Uploading RPMs to S3"
+s3cmd --acl-public sync ${REPO_DIR_LATEST} s3://${REPO_BUCKET}/
+
+# Public URL for the S3 bucket with our artifacts.
+MOCK_REPO_BASE_URL="http://osbuild-composer-repos.s3-website.us-east-2.amazonaws.com"
+
+# Full URL to the RPM repository after they are uploaded.
+REPO_URL=${MOCK_REPO_BASE_URL}/${REPO_DIR_LATEST}
+
+# Ammend repository file.
```
Comment (ondrejbudai) [inline schutzbot/prepare-rhel-nightly.sh:100]: `ammend` seems to be misspelled. Personally, I would go with just `create`. The same applies to the next line.

**Review by ondrejbudai — CHANGES_REQUESTED** [review]
Two nitpicks, otherwise this is looking good.

**Comment by teg** [comment]
I like the idea. Should we run this on the 8.3 branch (for 8.3 nightlies) and on master run against 8.4 nightlies?

File: schutzbot/prepare-rhel-nightly.sh (hunk 1 above)
Comment (atodorov) [reply schutzbot/prepare-rhel-nightly.sh:102]: Why do you think it isn't required ? We want to append the S3 repo to the same file b/c it already contains the nightly repos, see line 16.

**Comment by atodorov** [comment]
> I like the idea. Should we run this on the 8.3 branch (for 8.3 nightlies) and on master run against 8.4 nightlies?

At this point in time it is a bit late for the 8.3 nightlies IMO but this seems doable. There are -latest-RHEL-8.x links which could work with branches. Let's leave that as a follow up.

File: schutzbot/prepare-rhel-nightly.sh (hunk 1 above)
Comment (ondrejbudai) [reply schutzbot/prepare-rhel-nightly.sh:102]: I missed that, sorry, my fault!

File: schutzbot/prepare-rhel-nightly.sh (hunk 2 above)
Comment (ondrejbudai) [reply schutzbot/prepare-rhel-nightly.sh]: Yeah, `amend` makes sense.

**Comment by atodorov** [comment]
@tom, @ondrejbudai can we move forward on this PR ?

**Comment by atodorov** [comment]
@teg nightly job failed with:

```
sudo cp -a '/usr/share/tests/osbuild-composer/composer/*.toml' /etc/osbuild-composer/

[2020-10-07T22:05:59.255Z] cp: cannot stat '/usr/share/tests/osbuild-composer/composer/*.toml': No such file or directory
```

which I believe is related to your recent changes around tests. Do we want to safeguard with if conditions here or wait until 8.4 RPMs finally get rebased to the same version as upstream ?

**Comment by larskarlitski** [comment]
GitHub closed this pull request because I renamed osbuild's main branch of development to `main`, deleting `master`. If it is still relevant, please reopen it against the `main` branch. Thank you!

**Pull request closed without merging**